front/app/components/IdeaButton/index.tsx

```diff
@@ -97,41 +97,37 @@ const TooltipContentText = styled.div`
   }
 `;
 
-interface DataProps {
-  project: GetProjectChildProps;
-  phases: GetPhasesChildProps;
-  authUser: GetAuthUserChildProps;
-}
-
-interface InputProps extends Omit<ButtonProps, 'onClick'> {
+export interface Props extends Omit<ButtonProps, 'onClick'> {
   id?: string;
   projectId: string;
   latLng?: LatLng | null;
   inMap?: boolean;
   className?: string;
   participationContextType: IParticipationContextType;
-  buttonText?: MessageDescriptor;
   phase: IPhaseData | undefined;
+  participationMethod: Extract<
```

Comment:
Only allow the methods for which this button is rendered (we'll need these specific methods later, in a different file).